servus/serializable.h

```diff
@@ -77,49 +76,45 @@ class Serializable
      *
      * @return the binary representation of this object.
      */
-    Data toBinary() const { return _toBinary(); }
+    SERVUS_API Data toBinary() const;
 
     /**
      * Update this serializable from its JSON representation.
      * @return true on success, false on error.
      */
-    bool fromJSON( const std::string& json ) { return _fromJSON( json ); }
+    SERVUS_API bool fromJSON( const std::string& json );
 
     /** @return the JSON representation of this serializable. */
-    std::string toJSON() const { return _toJSON(); }
+    SERVUS_API std::string toJSON() const;
     //@}
 
     /** @name Change Notifications */
     //@{
     /** Function for change notification. */
 #ifdef SERVUS_USE_CXX03
-    typedef boost::function< void() > ChangeFunc;
+    typedef boost::function< void() > DeserializeFunc;
+    typedef boost::function< void() > SerializeFunc;
```

Comment:
While we are here I would also rename those to (De)SerializeCallback; they are not (de)serialization functions IMHO.